core/node/eth_sender/src/eth_tx_aggregator.rs

```diff
@@ -52,9 +59,15 @@ pub struct EthTxAggregator {
     aggregator: Aggregator,
     eth_client: Box<dyn BoundEthInterface>,
     config: SenderConfig,
-    timelock_contract_address: Address,
+    // The validator timelock address provided in the config.
+    // If the contracts have the same protocol version as the state transition manager, the validator timelock
+    // from the state transition manager will be used.
+    // The address provided from the config is only used when there is a discrepancy between the two.
+    // TODO(EVM-932): always fetch the validator timelock from L1, but it requires a protocol change.
```

Comment:
Just so I understand better: we could resolve TODO by always using chain's validator timelock but there is no easy way to query it from L1; so the task is mostly about changing contracts to support the query, right?

Reply:
right